KWZP2019/KWZP2019/EntranceControlStatisticsForm.cs

```diff
@@ -0,0 +1,223 @@
+                    select.Data_Kontroli)
+                    .ToList();
+
+                if(controlDates.Count == 0)
+                {
+                    MessageBox.Show("Brak kontroli wybranego półfabrykatu w bazie.", "Błąd", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+                }
+                else if(controlDates.Count == 1)
+                {
+                    datePickerDateFrom.Value = controlDates[0];
+                    datePickerDateTo.Value = controlDates[0];
+                }
+                else
+                {
+                    datePickerDateFrom.Value = controlDates.Min();
+                    datePickerDateTo.Value = controlDates.Max();
+                }
+                
+            }
+        }
+
+        // ==================================================
+
+        private void BtnShowHistograms_Click(object sender, EventArgs e)
+        {
+            if(selectedSemiFinishedId <= 0)
+            {
+                MessageBox.Show("Wybierz półfabrykat.", "Błąd", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+            else if(datePickerDateFrom.Value > datePickerDateTo.Value)
+            {
+                MessageBox.Show("Data \"od\" musi być większa od daty \"do\".", "Błąd", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            else
+            {
+                ShowHistograms(this.selectedSemiFinishedId, datePickerDateFrom.Value, datePickerDateTo.Value);
+            }
+        }
+
+        // ==================================================
+
+        private void BtnShowPieChart_Click(object sender, EventArgs e)
+        {
+            if (selectedSemiFinishedId <= 0)
+            {
+                MessageBox.Show("Wybierz półfabrykat.", "Błąd", MessageBoxButtons.OK, MessageBoxIcon.Warning);
+            }
+            else if (datePickerDateFrom.Value > datePickerDateTo.Value)
+            {
+                MessageBox.Show("Data \"od\" musi być większa od daty \"do\".", "Błąd", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            }
+            else
+            {
+                ShowPieChart(this.selectedSemiFinishedId, datePickerDateFrom.Value, datePickerDateTo.Value);
+            }
+        }
+
+        // ==================================================
+
+        private void DataGridViewSemiFinished_RowHeaderMouseDoubleClick(object sender, DataGridViewCellMouseEventArgs e)
+        {
+            DataGridViewRow selectedRow = dataGridViewSemiFinished.SelectedRows[0];
+            selectedSemiFinishedId = (int)selectedRow.Cells["Id"].Value;
+        }
+
+        // ==================================================
+
+        private void ShowHistograms(int idSemiFinished, DateTime dateFrom, DateTime dateTo)
+        {
+            ProcessStartInfo processStartInfo = new ProcessStartInfo();
+            processStartInfo.FileName = @"C:\Users\bartl\AppData\Local\Programs\Python\Python37-32\python.exe";
```

Comment:
taka ścieżka nie zadziała

Reply:
U mnie działa. Na uczelnianych komputerach trzeba by było ją zmienić. Zostanie to usunięte i ewentualnie zrobione za pomocą LiveChart albo czegoś innego.

Reply:
w takim razie do usunięcia z tego PR